### ISSUE TITLE
[Feature Request] Support the parameter "shallow_since"

### ISSUE THREAD

**Issue by suzuki-shunsuke** [guitarrapc/git-shallow-clone-orb#9]
`git clone` supports the option [--shallow-since](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---shallow-sinceltdategt).

```
--shallow-since=<date>
Create a shallow clone with a history after the specified time.
```

To support the option, I propose to add the optional parameter `shallow_since`.
By default this parameter is empty and `--shallow-since` isn't set, so this change keeps the compatibility.
If this parameter is set, the parameter `--shallow-since` is set and the parameter `depth` is ignored.

What do you think?

Thank you.

**Comment by guitarrapc** [comment]
seems nice, but if I add support for this particular options future will become too many arguments.
Is adding support for `OPTIONS` would be more user-developer friendly?

```
  options:
    description: >
      git options you want to add. like `--shallow-since "5 days ago"`
    type: string
    default: ""
```

**Comment by suzuki-shunsuke** [comment]
Thank you for your reply.
Cool. 

Then I propose that when the option `clone_options` is set the parameter `depth` is ignored because `--depth` argument may conflict other argument.
If users want to set the depth parameter along with `clone_options`,
they can set `--depth` argument in `clone_options`.

Thank you.

**Comment by guitarrapc** [comment]
released 1.2.0.

> https://circleci.com/orbs/registry/orb/guitarrapc/git-shallow-clone
